homeassistant/components/brunt/cover.py

```diff
@@ -2,84 +2,99 @@
 from __future__ import annotations
 
 import logging
+from typing import Any
 
-from brunt import BruntAPI
-import voluptuous as vol
+from aiohttp.client_exceptions import ClientResponseError
+from brunt import BruntClientAsync, Thing
 
 from homeassistant.components.cover import (
     ATTR_POSITION,
     DEVICE_CLASS_WINDOW,
-    PLATFORM_SCHEMA,
     SUPPORT_CLOSE,
     SUPPORT_OPEN,
     SUPPORT_SET_POSITION,
     CoverEntity,
 )
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_PASSWORD, CONF_USERNAME
-import homeassistant.helpers.config_validation as cv
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    ATTR_REQUEST_POSITION,
+    ATTRIBUTION,
+    CLOSED_POSITION,
+    DOMAIN,
+    FAST_INTERVAL,
+    OPEN_POSITION,
+    REGULAR_INTERVAL,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 COVER_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
 
-ATTR_REQUEST_POSITION = "request_position"
-NOTIFICATION_ID = "brunt_notification"
-NOTIFICATION_TITLE = "Brunt Cover Setup"
-ATTRIBUTION = "Based on an unofficial Brunt SDK."
-
-CLOSED_POSITION = 0
-OPEN_POSITION = 100
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
-)
 
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
Please implement a coroutine function `async_setup_platform` and import the legacy config YAML to a config flow.